tcms/kiwi_auth/tests/test_admin.py

```diff
@@ -124,6 +124,22 @@ def test_admin_can_add_new_users(self):
             get_user_model().objects.filter(username="added-by-admin").exists()
         )
 
+    def test_admin_can_delete_itself(self):
+        self.client.login(  # nosec:B106:hardcoded_password_funcarg
+            username=self.admin.username, password="admin-password"
+        )
+        response = self.client.get(
+            reverse("admin:auth_user_delete", args=[self.admin.pk])
+        )
+        self.assertContains(response, _("Yes, I'm sure"))
+        response = self.client.post(
+            reverse("admin:auth_user_delete", args=[self.admin.pk]),
+            {"post": "yes"},
+            follow=True,
+        )
+
+        self.assertRedirects(response, reverse("core-views-index"))
```

Comment:
This fails because upon deletion the user doesn't exist anymore, so the browser session becomes unauthenticated. When an unauthenticated browser hits `/` it is then redirected to `/accounts/login/`, see the `next` parameter which tells the login view to perform another redirect back to the dashboard page.